File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/multiline/comment/direct/Java/Python/s831281137_commented.py
import sys

def main():
    input = sys.stdin.read().split()
    index = 0
    
    N = int(input[index])
    W = int(input[index + 1])
    index += 2
    
    v = [0] * (N + 1)
    w = [0] * (N + 1)
    
    for i in range(1, N + 1):
        v[i] = int(input[index])
        w[i] = int(input[index + 1])
        index += 2
    
    sumV = [[0] * (W + 1) for _ in range(N + 1)]
    
    for i in range(1, N + 1):
        for i_sum in range(w[i]):
            sumV[i][i_sum] = sumV[i - 1][i_sum]
        
        for i_sum in range(w[i], W + 1):
            sumV[i][i_sum] = max(sumV[i - 1][i_sum], sumV[i - 1][i_sum - w[i]] + v[i])
    
    print(sumV[N][W])

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/uncomment/direct/Java/C/s831281137_commented.c
#include <stdio.h>
#include <string.h>

int main() {
    int N, W;
    scanf("%d %d", &N, &W);
    
    int v[N+1], w[N+1];
    for (int i = 1; i <= N; i++) {
        scanf("%d %d", &v[i], &w[i]);
    }
    
    int sumV[N+1][W+1];
    memset(sumV, 0, sizeof(sumV));
    
    for (int i = 1; i <= N; i++) {
        for (int i_sum = 0; i_sum < w[i]; i_sum++) {
            sumV[i][i_sum] = sumV[i-1][i_sum];
        }
        
        for (int i_sum = w[i]; i_sum <= W; i_sum++) {
            sumV[i][i_sum] = (sumV[i-1][i_sum] > (sumV[i-1][i_sum - w[i]] + v[i])) ? 
                             sumV[i-1][i_sum] : (sumV[i-1][i_sum - w[i]] + v[i]);
        }
    }
    
    printf("%d\n", sumV[N][W]);
    return 0;
}
// End of Code
